src/lib/core/overlay/position/viewport-ruler.ts

```diff
@@ -55,4 +66,13 @@ export class ViewportRuler {
 
     return {top, left};
   }
+
+  /**
+   * @internal
+   * Updates the cached client rectangle of the document element.
+   */
+  updateDocumentRectangle() {
```

Comment:
Yep; using `@internal` causes issues with the Closure Compiler, which is why we have the `@docs-private` annotation. It's easier to just use an underscore prefix, though (without the `private` access modifier if it's not actually private).

I would also call this something like `cacheViewportGeometry`; using the "update" verb implies that calling this function will enact some change on the document rectangle instead of reading (and caching) it.